Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,148 @@ This document includes four main sections:
+to commit the transaction::
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   con.commit()
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
+Execute a query to verify that the data was inserted correctly.
+Use the now familiar :meth:`~Cursor.execute` method on ``con``,
+store the result in ``res``,
+and call :meth:`~Cursor.fetchall` on ``res`` to fetch all rows::
+
+   >>> res = cur.execute("SELECT Score FROM Movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
+
+The result is a :class:`list` with two rows of data,
+each a one-item :class:`tuple`,
+the single item representing the column queried.
+
+Now, insert three more rows by calling
+:meth:`~Cursor.executemany` on the cursor object,
+once again committing the transaction by calling :meth:`~Connection.commit`
+on the connection object::
+
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO Movie VALUES(?, ?, ?)", data)
+   con.commit()
 
-Notice that we used ``?`` placeholders to bind *data* to the query.
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
 to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
+See :ref:`sqlite3-placeholders` for more details.
 
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
+Verify that data has been committed and written to disk by
+closing the connection and opening a new one,
+creating a new cursor,
+and then executing a ``SELECT`` query to read from the database::
 
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+   >>> con.close()
+   >>> con2 = sqlite3.connect("tutorial.db")
+   >>> cur2 = con2.cursor()
+
+This time, retrieve the rows by iterating over the result
+of the ``SELECT`` query::
+
+   >>> for row in cur2.execute("SELECT Year, Title FROM Movie ORDER BY Year"):
    ...     print(row)
+   ...
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
 
-   ('2006-01-05', 'BUY', 'RHAT', 100, 35.14)
-   ('2006-03-28', 'BUY', 'IBM', 1000, 45.0)
-   ('2006-04-06', 'SELL', 'IBM', 500, 53.0)
-   ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0)
+Each row is now a two-item :class:`!tuple`.
```

Comment:
It would be better to be more explicit on why it's a two-item tuple and what the two items are, so that readers can build a connection between the query and the result::
```suggestion
Each row is a ``(Year, Title)`` :class:`!tuple`.
```
If you prefer something more verbose: `Each row is a two-item tuple that contains the selected columns: ``Year`` and ``Title``.`

Also I've seen two tuples linked, and this third isn't.

Reply:
+1

I actually had it more verbose in a revision I did not push.

Reply:
We can link the third tuple as well. Perhaps we should unlink the second.